src/ethereum/api/graphClient.ts

```diff
@@ -18,3 +22,35 @@ export function createClient(url: string) {
 
   return client
 }
+
+async function fetchWithTimeout(uri: string, options: RequestInit) {
+  const timeout = Number(env.get('GRAPH_QUERY_TIMEOUT', 10000))
+  const logger = createConsoleLogComponent().getLogger('Fetch GraphAPI')
+
+  try {
+    const controller = new AbortController()
+
+    const id = setTimeout(() => {
+      controller.abort()
+    }, timeout)
+
+    const response = await nodefetch(uri, {
+      ...options,
+      signal: controller.signal as RequestInit['signal'],
+    })
+
+    clearTimeout(id)
+
+    return response
+  } catch (error) {
+    if (isErrorWithMessage(error) && error.message.includes('aborted')) {
+      const requestBody = JSON.parse(options.body as string)
+      logger.error(
+        `Timeout fetching the subgraph: ${uri}/${requestBody.operationName}`
+      )
+      throw new Error(error.message)
```

Comment:
LOL, creating a new error fixed the issue?

Reply:
yeap, for some reason, the aborted error wasn't captured by the apollo client :/